Services/Concurrency/RateLimiting.cs

```diff
@@ -58,6 +60,50 @@ public class RateLimiting : IRateLimiting
             log.Info("Rate limiting started. This message should appear only once in the logs.");
         }
 
+        public Models.Simulation.SimulationRateLimits GetRateLimits()
+        {
+            return new Models.Simulation.SimulationRateLimits
+            {
+                ConnectionsPerSecond = this.connections.EventsPerTimeUnit,
```

Comment:
Counters are only set SetCounters for condition > 0, so this.connections can be null if the condition is not true. This will throw an exception here.
Same for other counters.